flink-core/src/main/java/org/apache/flink/configuration/NettyShuffleEnvironmentOptions.java

```diff
@@ -54,6 +54,28 @@
 			.withDescription("Enable SSL support for the taskmanager data transport. This is applicable only when the" +
 				" global flag for internal SSL (" + SecurityOptions.SSL_INTERNAL_ENABLED.key() + ") is set to true");
 
+	/**
+	 * Boolean flag indicating whether the shuffle data will be compressed or not.
+	 *
+	 * <p>Note: Data is compressed per buffer (may be sliced buffer in pipeline mode) and compression can incur extra
+	 * CPU overhead so it is more effective for IO bounded scenario when data compression ratio is high.
+	 */
+	public static final ConfigOption<Boolean> DATA_COMPRESSION_ENABLED =
+		key("taskmanager.data.compression.enabled")
```

Comment:
Have you thought about providing just a single option `taskmanager.network.compression`, with default value `"NONE"` and one more currently supported `"LZ4"`?

On another note... what if someone wants to use compression just for `BlockingBoundedPartitions`? Actually that could be a sane/valid default: `LZ4` for partitions being written to disk and `NONE` for pipelined?

```
taskmanager.network.bounded-blocking-subpartition.compression: NONE/LZ4 (default LZ4?)
taskmanager.network.compression: NONE/LZ4 (defaul NONE?)
```

Reply:
I am a little afraid that when we make it pluggable in the future, the "LZ4" config value can be  dropped and users already using compression may have to change their configurations. Currently, the codec config option is not documented and we can change it to any other values in the future when we make it pluggable in the future.

Reply:
If that happens we could always map `LZ4` to some new value. 

I talked with @NicoK and we both slightly would prefer to have fewer configuration parameters (`taskmanager.network.blocking-shuffle.compression: NONE/LZ4` and `taskmanager.network.pipelined-shuffle.compression: NONE/LZ4`). However it's not a blocker for us if you have a strong opposite feeling.

If we moved to `taskmanager.network.blocking-shuffle.compression: NONE/LZ4` and `taskmanager.network.pipelined-shuffle.compression: NONE/LZ4`, the problem might be to support a use case when user configures different compression algorithms for `pipelined` and `blocking`? If that can not be easily supported, let's just drop the discussion and leave it as it is now.

Reply:
I also prefer to one option instead of two. `NONE` means not enabling compression and we can also support other compression algorithms for pipelined/blocking if setting other specific values.

Reply:
Thanks for the comments. After reconsidering the problem, I think the user should only need to enable/disable compression and decide which compression plugin to use in the future. All other strategies should be left to the plugin, including which compression algorithm to use (may be different for pipelined and blocking), enable other compression optimization configurations and so on. So I would just leave it as it is now.